### ISSUE TITLE
Missing font-locking

### ISSUE THREAD

**Issue by manuel-uberti** [clojure-emacs/clojure-mode#474]
## Expected behavior
Clojure own functions/forms correctly highlighted. 

## Actual behavior
As of now, `if`, `let`, `loop` and `recur` (for instance) are correctly highlighted, but `select-keys`, `map`,
 `get-in` and `concat` (the first ones I catch) are not.

![screenshot](https://user-images.githubusercontent.com/8199224/37390560-ceac994e-2768-11e8-8990-2855c8ecbce8.png)

## Steps to reproduce the problem
- open a .clj file with `clojure-mode` installed
- write `(map inc [1 2 3])`

### clojure-mode version information
```
clojure-mode (version 5.7.0-snapshot)
```

### Emacs version

GNU Emacs 27.0.50 (build 1, x86_64-debian-linux-gnu, GTK+ Version 3.18.9) of 2018-03-14

### Operating system

Ubuntu 16.04

**Comment by manuel-uberti** [comment]
![screenshot](https://user-images.githubusercontent.com/8199224/37395866-9ad27026-2777-11e8-88be-1f8ab84f277e.png)

Namespaces have problem has well.

**Comment by bbatsov** [comment]
This is some regression from https://github.com/clojure-emacs/clojure-mode/pull/462 

@Bost Mind checking what went wrong?

**Comment by Olical** [comment]
And namespace keywords only highlight the second colon, the first is left white.

**Comment by Bost** [comment]
@Olical the highlighting of namespaced keywords is a regression from #462 indeed. (I'll fix it)

@manuel-uberti @bbatsov font-locking of `select-keys map get-in concat` hasn't changed since 5.6.1. Anyway it can be easily changed by applying this patch:
```
index 073e287..0d28c63 100644
--- a/clojure-mode.el
+++ b/clojure-mode.el
@@ -789,6 +789,7 @@ any number of matches of `clojure--sym-forbidden-rest-chars'."))
          "("
          (regexp-opt
           '("def" "do" "if" "let" "let*" "var" "fn" "fn*" "loop" "loop*"
+            "select-keys" "map" "get-in" "concat"
             "recur" "throw" "try" "catch" "finally"
             "set!" "new" "."
             "monitor-enter" "monitor-exit" "quote") t)
```

**Comment by manuel-uberti** [comment]
I see, but even `assoc`, `assoc-in`, `contains?` are not highlighted now. Should every function/macro/form be added then?

**Comment by manuel-uberti** [comment]
Also, user-defined macros are not highlighted either.

![screenshot](https://user-images.githubusercontent.com/8199224/37411596-6f201df8-27a3-11e8-8c4c-549bd48c0219.png)

**Comment by bbatsov** [comment]
> @manuel-uberti @bbatsov font-locking of select-keys map get-in concat hasn't changed since 5.6.1. Anyway it can be easily changed by applying this patch:

@Bost Seems I was unclear. Sorry about this. Have you tried whether the dynamic font-locking in CIDER was affected by your changes?

**Comment by manuel-uberti** [comment]
> Those are getting highlighted by CIDER, not clojure-mode. Perhaps @Bost's changes to the font-locking regexps affected CIDER's font-locking.

FWIW, the screenshot with `test-macro` comes from a .clj file with CIDER active.

**Comment by manuel-uberti** [comment]
@Bost is there anything I can do to help fixing this?

**Comment by Bost** [comment]
> @Bost Seems I was unclear. Sorry about this. Have you tried whether the dynamic font-locking in CIDER was affected by your changes?

Uhm, how that works? I mean I copy-pasted the content of clojure-mode/test.clj to *cider-scratch* and it looks fine (Changes from my latest PR #475 were included).

**Comment by Bost** [comment]
Unfortunately the `spacemacs//find-ert-test-buffer` doesn't work for my any more so I let the travis do the testing for me. That' why you see here the PR retries.

**Comment by bbatsov** [comment]
@Bost See http://cider.readthedocs.io/en/latest/configuration/#dynamic-syntax-highlighting

**Comment by Bost** [comment]
@bbatsov I can't get the dynamic-syntax-highlighting working... hmm.
I compile my emacs from the source. I tried branches origin/emacs-26, origin/emacs-25, clojure-mode master and also 5.6.1 and I also tried to run `.src/emacs --no-init-file` and then installed cider and clojure-mode from melpa-stable. And nothing... :neckbeard:

**Comment by bbatsov** [comment]
Seems to work just fine for me with the old version of clojure-mode. I just start CIDER and it's there. For you there's 0 difference before and after?

**Comment by bbatsov** [comment]
Btw, I wonder if your color theme makes it harder to spot the differences or something. Also - which CIDER are you using?

**Comment by Bost** [comment]
I was launching emacs with `--no-init-file` and using `melpa-stable` (and then `M-x cider-jack-in` and `M-x cider-scratch`) so everything it was using default vals basically everywhere.... but(!) the `dynamic-syntax-highlighting` doesn't appear until you actually EVALUATE some sexp containing something dynamically highlightable!!! Grrr! :rage4: Ok. Now I can do the actual problem analysis...

**Comment by Bost** [comment]
... and now I'm confirming this issue is a regression from #462 indeed.

**Comment by bbatsov** [comment]
@Bost Any progress with solving it? Breaking the dynamic font-locking is much bigger than fixing ns font-locking, so without some solution for this soon I'll have to revert the previous commits.

**Comment by Bost** [comment]
I think it's better to revert it for the moment. (Although I'll be having some time this week to work on it.)

**Comment by Bost** [comment]
Huh, I think the dynamic syntax highlighting is fixed. (It was easier than expected). Please have a look at the PR 475.
@bbatsov Regarding the revert of #462 - since the problem appears to be a simple one and I do have time, please consider postponing the decision until the next week. Thx.

**Comment by bbatsov** [comment]
Fixed in #475.
